api/v1alpha1/hostedcluster_types.go

```diff
@@ -21,6 +21,11 @@ type HostedClusterSpec struct {
 	// workers. It should have an ".dockerconfigjson" key containing the pull secret JSON.
 	PullSecret corev1.LocalObjectReference `json:"pullSecret"`
 
+	// AuditWebhook contains metadata for configuring an audit webhook
+	// endpoint for a cluster to process cluster audit events
+	// +optional
+	AuditWebhook corev1.LocalObjectReference `json:"auditWebhook,omitempty"`
```

Comment:
What are the conventions here on pointer vs. non-pointer for this optional fields of this particular type?

Reply:
I am not sure what you mean by conventions. When it's not specified the name comes back as an empty string currently. When specified: it is populated.

Reply:
I think I see what you mean: I standardized on "pointer" for optional

Reply:
What I meant was, I thought by upstream convention optional fields should be pointers in general, but I didn't know if there were exceptions for some value types in conjunction with `omitempty`. I notice signingKey is also optional but isn't a pointer. Perhaps @openshift/api-reviewers can clarify?

Reply:
Pinging @sjenning